### DISCUSSION TITLE
Question:

### DISCUSSION THREAD

**Discussion by boyangsic** [FunkinCrew/Funkin#1570] [Q&A] [answered]
-

**Comment by ghost** [comment]
IDK maybe?

**Comment by xenten9** [comment]
I mean realistically speaking there is no copyrighting the concept of a game; However, you may be required to make your own engine and code if you wanted to sell it commercially, and if your game is going to be an open source derivation of the game Friday Night Funkin' then you need to give credit and adopt the same license if this game's license allows for modifications under the same license. What I think you should do is look into making your own engine as that way you are more familiar and better able to debug it, although it is understandable why that may be outside of the scope of what you can or are willing to do.

**Comment by boyangsic** [comment]
i use unity to make games, is it possible?

**Comment by xenten9** [comment]
The game as it stands is written in Haxe for Flixel which is in general a modern way of doing flash like games. However Unity is its own separate thing, so you probably would struggle to borrow code given most code for Unity is written in C#/C++. For the most part you could totally make your own rhythm game without any problems. You would just be responsible for ensuring your don't break copyright/trademark laws, and you would be responsible for coding the game from the ground up in Unity. In my personal opinion unless you have worked on similar projects before with a team or have low aspirations for your final product, I would not go into this venture. Game dev takes a lot of hours and as a single person that translates into many months and sometimes years, so just keep that in mind. Unity is a powerful tool though and realistically you could probably get a decent albeit needing a lot of improvements rhythm game. The main thing that separates good and bad rhythm games at least on the low level is how well they can be adjusted to match up the video audio and input recognition. If you can handle all of that then realistically you are already a long ways towards making a rhythm game.

**Comment by isakube** [comment]
> i use unity to make games, is it possible?

copy Quaver instead then lol

(don't)

**Comment by SpGerg** [comment, marked as answer]
I have already tried writing this game

https://www.youtube.com/watch?v=6BKSWZW5UAU&ab_channel=SpGergGames

**Comment by RevixDev** [comment]
If you are going to create a game based of the Friday Night Funkin' base engine. Then I would say atleast credit idk? And make sure its open source

**Reply by RevixDev** [reply]
Otherwise NinjaMuffin will kick you to brazil
